src/k8s/pkg/k8sd/app/hooks_remove.go

```diff
@@ -107,6 +103,10 @@ func (a *App) onPreRemove(ctx context.Context, s state.State, force bool) (rerr
 		log.Error(err, "Failed to retrieve cluster config")
 	}
 
+	log.Info("Cleaning up k8s-dqlite directory")
+	if err := os.RemoveAll(snap.K8sDqliteStateDir()); err != nil {
+		return fmt.Errorf("failed to cleanup k8s-dqlite state directory: %w", err)
```

Comment:
Is this a best effort cleanup, if so I guess we should just `log.Error` instead of failing the whole hook right?

Reply:
Indeed, you are right. We were originally removing the error as well, so this was moved as is. Corrected.

Though, `onPreRemove` returns an error. If we're only doing best effort here, what error should we eventually return, considering the fact that there are many error cases?

Reply:
That is a good question. What could be a better way is maybe to use an error group, run through all steps and return error(s) through there? @bschimke95